core/utils.c

```diff
@@ -149,8 +151,11 @@ int utils_textToFloat(const uint8_t * buffer,
         i = 0;
     }
 
+    /* Must have a decimal digit first after optional sign */
+    if (i >= length || buffer[i] < '0' || buffer[i] > '9') return 0;
```

Comment:
Using `isdigit()`/`#include <ctype.h>` makes sense here too.

Reply:
I'll change these as they are new with this pull request, but just grepping for "'0'" shows many more places that isdigit() could be used.